src/dmd/traits.d

```diff
@@ -1418,18 +1418,29 @@ Expression semanticTraits(TraitsExp e, Scope* sc)
         auto o1 = (*e.args)[1];
 
         FuncDeclaration fd;
-        TypeFunction tf = toTypeFunction(o, fd);
-
+        Expression exp = isExpression(o);
+        CallExp ce;
         ParameterList fparams;
-        if (tf)
-            fparams = tf.parameterList;
-        else if (fd)
-            fparams = fd.getParameterList();
+
+        if(exp)
+            ce = exp.isCallExp();
+
+        TypeFunction tf;
+        if(ce)
+            fparams = ce.f.getParameterList();
         else
         {
-            e.error("first argument to `__traits(getParameterStorageClasses, %s, %s)` is not a function",
-                o.toChars(), o1.toChars());
-            return ErrorExp.get();
+            tf = toTypeFunction(o, fd);
+            if (tf)
+                fparams = tf.parameterList;
+            else if (fd)
+                fparams = fd.getParameterList();
+            else
+            {
+                e.error("first argument to `__traits(getParameterStorageClasses, %s, %s)` is not a function",
+                    o.toChars(), o1.toChars());
```

Comment:
Please adjust the error now that it can also be a call expression.

Reply:
Done